app/controllers/users/verify_password_controller.rb

```diff
@@ -29,12 +27,19 @@ def update
     private
 
     def confirm_personal_key
-      account_recovery = user_session[:account_recovery]
-      redirect_to root_url unless account_recovery[:personal_key]
+      return if reactivate_account_session.personal_key?
+      redirect_to root_url
```

Comment:
Sorry for the delay in reviewing this. Looks good to me overall. Looks like we are not testing this line according to Code Climate.